File: programmers/pg_lv1_약수의합.js
function solution(n) {
  let half = Math.ceil(n / 2);
  let sum = n;

  if (n == 0) return 0;
  if (n == 1) return 1;

  for (i = 1; i <= half; i++) {
    if (n % i == 0) sum += i;
  }
  return sum;
}

File: python/programmers/pg_lv1_약수의합.py
from math import sqrt

def solution(n):
    if n < 2: return n
    
    totalSum = 0
    sqrtNum = int(sqrt(n))

    for i in range(1, sqrtNum + 1):
        if n % i == 0:
            partner = n // i
            totalSum += i if i == partner else (i + partner)
            
    return totalSum